component/attr_cache/attr_cache.go

```diff
@@ -184,14 +185,17 @@ func (ac *AttrCache) deleteDirectory(path string, time time.Time) {
 	// Add a trailing / so that we only delete child paths under the directory and not paths that have the same prefix
 	prefix := dirToPrefix(path)
 
-	for key, value := range ac.cacheMap {
-		if strings.HasPrefix(key, prefix) {
-			value.markDeleted(time)
-		}
+	//get attrCacheItem
+	toBeDeleted, err := ac.cacheMap.get(prefix)
+
+	// delete the path itself and children.
+	if err != nil {
+		log.Err("could not find the cache map item due to the following error: ", err)
```

Comment:
If we know that the error will always be about not finding the directory, then this would make sense. I feel like there can be more than one kind of error and so I left it more general for this reason and this includes the error telling us that the item or folder not being found.

Reply:
In this case, you wrote the get() function, so it only returns an error when the key is not in the tree.